package Egametang;

message EventNode
{
	required int32 type = 2;
	// 条件需要的参数
	repeated int32 args = 3;
	// 包含多个子节点
	repeated EventNode nodes = 4;
};

message EventConf
{
	// 事件类型
	required int32 type = 1;
	// 条件
	required ConditionConf condition = 2;
	// 动作
	required ActionConf action = 3;
};

message ConditionConf
{
	optional EventNode node = 1;
};

message ActionConf
{
	optional EventNode node = 1;
};